lib/docsize_cache.js

```diff
@@ -0,0 +1,108 @@
+var LRU = Npm.require('lru-cache');
+var crypto = Npm.require('crypto');
+
+DocSzCache = function (maxItems, maxValues) {
+  this.items = new LRU({max: maxItems});
+  this.maxValues = maxValues;
+}
+
+DocSzCache.prototype.getSize = function (coll, query, opts, data) {
+  // If the dataset is null or empty we can't calculate the size
+  // Do not process this data and return 0 as the document size.
+  if (!(data && data.length)) {
+    return 0;
+  }
+
+  var key = this.getKey(coll, query, opts);
+  var item = this.items.get(key);
+
+  if (!item) {
+    item = new DocSzCacheItem(this.maxValues);
+    this.items.set(key, item);
+  }
+
+  if (this.needsUpdate(item)) {
+    var size = JSON.stringify(data[0]).length;
```

Comment:
Shall we use the mean value of all items? I think using the mean will be fair in this case and calculating the mean is also quite easy (stringify the whole array and divide by array length).